### DISCUSSION TITLE
application.js is loading twice in Firefox

### DISCUSSION THREAD

**Discussion by rachitpant** [ElMassimo/vite_ruby#209] [Q&A] [answered]
I initialized a new project with vue2 and composition api and pinia instead of vuex , I also used your other helpful package to generate js apis from rails routes. 
In this new project , I am getting couple of different (but similar) warnings on the browser console which i am not getting in other projects.

<img width="1662" alt="Screen Shot 2022-04-28 at 12 01 16 PM" src="https://user-images.githubusercontent.com/1474415/165795592-a10d46a2-5115-4848-ab5e-8b55d9e21222.png">

It doesn't break anything , but can you think of anything that could be causing this.

I also noticed only in this app , there are multiple requests(two) to `application.js` in browser network console.
Might be unrelated , but i thought i should mention this.

<img width="1264" alt="image" src="https://user-images.githubusercontent.com/1474415/165796112-e84e2b11-7e34-4b45-a14e-d1c48142a3a5.png">

This is a brand new project with vite_ruby , as opposed to other projects which i migrated from webpacker to vite.

**Comment by ElMassimo** [comment, marked as answer]
Hi Rachit!

If it only happens in Firefox it might be related to https://github.com/vitejs/vite/issues/5532.

If `application.js` is being requested twice, it's probably related to the setup.
Check that you are not accidentally injecting `vite_javascript_tag` several times in different partials/layouts, or look at each request in the inspector to find the initiator in the _Network_ tab.

The warning about a preloaded resource is likely caused by the previous _Loading failed_ warning.
You might want to verify that the CSP settings in Rails are not [as strict during development](https://github.com/ElMassimo/pingcrm-vite/blob/main/config/initializers/content_security_policy.rb#L15-L20), as that could be blocking the JS from being loaded.

**Reply by rachitpant** [reply]
I checked the application setup and it seems to be fine ( single `vite_javascript_tag` and also the CSP looks good ). Just now I checked on chrome and there are no such warnings and also the files are not requested multiple times. 
Wondering if its related to [vite issue 5532](https://github.com/vitejs/vite/issues/5532)

**Reply by ElMassimo** [reply]
Definitely seems related to https://github.com/vitejs/vite/issues/5532.

I'd suggest using the `build.polyfillModulePreload: false` workaround, if that's viable for your use case.
